pkg/kubelet/kubelet.go

```diff
@@ -2421,6 +2424,46 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}
 }
 
+func (kl *Kubelet) syncOSAndArchLabelsOnce() (*v1.Node, error) {
+	node, err := kl.GetNode()
+	if err != nil {
+		return nil, fmt.Errorf("error getting node: %v", err)
+	}
+	newNode := node.DeepCopy()
+	// Set the OS label if there is mismatch
+	osName, ok := newNode.Labels[v1.LabelOSStable]
+	if !ok || osName != sysruntime.GOOS {
+		if len(newNode.Labels) == 0 {
+			newNode.Labels = make(map[string]string)
+		}
+		newNode.Labels[v1.LabelOSStable] = sysruntime.GOOS
+	}
+	// Set the arch if there is a mismatch
+	arch, ok := newNode.Labels[v1.LabelArchStable]
+	if !ok || arch != sysruntime.GOARCH {
+		if len(newNode.Labels) == 0 {
+			newNode.Labels = make(map[string]string)
+		}
+		newNode.Labels[v1.LabelArchStable] = sysruntime.GOARCH
+	}
+	updatedNode, patchBytes, err := nodeutil.PatchNodeWithStatusAndLabels(kl.kubeClient.CoreV1(), node.Name, node, newNode)
+	if err != nil {
+		return nil, fmt.Errorf("error patching while updateing node labels: %v", err)
+	}
+	klog.V(6).InfoS("Patching successful", "patchBytes", patchBytes)
+	return updatedNode, nil
+}
+
+func (kl *Kubelet) syncOSAndArchLabels() {
+	for {
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Why are we writing our own sleep and not using one of the utility functions in `wait` for this?